examples/models/llama2/aqlm/CmakeLists.txt

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+cmake_minimum_required(VERSION 3.19)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+
+if(NOT PYTHON_EXECUTABLE)
+  set(PYTHON_EXECUTABLE python3)
+endif()
+
+# Source root directory for executorch.
+if(NOT EXECUTORCH_ROOT)
+  set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../../../..)
+endif()
+
+set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+
+include(${EXECUTORCH_ROOT}/build/Utils.cmake)
+include(${EXECUTORCH_ROOT}/build/Codegen.cmake)
+
+#
+# The `_<target>_srcs` lists are defined by including ${EXECUTORCH_SRCS_FILE}.
+#
+set(EXECUTORCH_SRCS_FILE
+    "${CMAKE_CURRENT_BINARY_DIR}/../../../../executorch_srcs.cmake"
+)
+
+extract_sources(${EXECUTORCH_SRCS_FILE})
+
+include(${EXECUTORCH_SRCS_FILE})
+
+# Let files say "include <executorch/path/to/header.h>".
+set(_common_include_directories ${EXECUTORCH_ROOT}/..)
+
+# Custom op libraries
+set(aqlm_libs executorch_no_prim_ops)
+list(APPEND aqlm_libs pthreadpool)
+list(APPEND aqlm_libs cpuinfo)
+list(APPEND aqlm_libs cpublas)
+list(APPEND aqlm_libs eigen_blas)
+
+set(_aqlm__srcs examples/models/llama2/aqlm/lut_kernel.h examples/models/llama2/aqlm/lut_kernel.cpp)
+list(TRANSFORM _aqlm__srcs PREPEND "${EXECUTORCH_ROOT}/")
+
+
+
+message("HERE: AQLM SOURCES: ${_aqlm__srcs}")
+
+# TODO: Consider moving xnnpack/threadpool in a separate lib since it's now used
+# by custom ops too.
+if(NOT EXECUTORCH_BUILD_XNNPACK)
+  list(
+    APPEND
+    _aqlm__srcs
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../../backends/xnnpack/threadpool/threadpool.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../../backends/xnnpack/threadpool/threadpool_guard.cpp"
+  )
+else()
+  list(APPEND aqlm_libs xnnpack_backend)
+endif()
+
+# find_package(OpenMP REQUIRED)
+# list(APPEND aqlm_libs OpenMP::OpenMP_CXX)
+# list(APPEND aqlm_libs omp)
+
+add_library(aqlm ${_aqlm__srcs})
```

Comment:
A library containing core AQLM ops and a `EXECUTORCH_LIBRARY` for their automatic registration into executorch runtime when linked with `target_link_options_shared_lib`.